pkg/reconciler/reconciler.go

```diff
@@ -167,6 +173,79 @@ func NewBase(opt Options, controllerAgentName string) *Base {
 	return base
 }
 
+// NewBase instantiates a new instance of Base implementing
+// the common & boilerplate code between our reconcilers.
+func NewInjectionBase(ctx context.Context, controllerAgentName string, cmw configmap.Watcher) *Base {
+	// Enrich the logs with controller name
+	logger := logging.FromContext(ctx).
+		Named(controllerAgentName).
+		With(zap.String(logkey.ControllerType, controllerAgentName))
+
+	kubeClient := kubeclient.Get(ctx)
+
+	recorder := GetEventRecorder(ctx)
+	if recorder == nil {
+		// Create event broadcaster
+		logger.Debug("Creating event broadcaster")
+		eventBroadcaster := record.NewBroadcaster()
+		watches := []watch.Interface{
+			eventBroadcaster.StartLogging(logger.Named("event-broadcaster").Infof),
+			eventBroadcaster.StartRecordingToSink(
+				&typedcorev1.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")}),
+		}
+		recorder = eventBroadcaster.NewRecorder(
+			scheme.Scheme, corev1.EventSource{Component: controllerAgentName})
+		go func() {
+			<-ctx.Done()
+			for _, w := range watches {
+				w.Stop()
+			}
+		}()
+	}
+
+	statsReporter := GetStatsReporter(ctx)
+	if statsReporter == nil {
+		logger.Debug("Creating stats reporter")
+		var err error
+		statsReporter, err = NewStatsReporter(controllerAgentName)
+		if err != nil {
+			logger.Fatal(err)
+		}
+	}
+
+	base := &Base{
+		KubeClientSet:     kubeClient,
+		EventingClientSet: eventingclient.Get(ctx),
+		//ApiExtensionsClientSet: opt.ApiExtensionsClientSet,
```

Comment:
Remove.

Reply:
I think we want to add this to the hand-rolled stuff in `knative/pkg`?

Reply:
we need it for other things in this project, just not the sources.

Reply:
It will have to happen when we move over the eventing side.